articles/055_ros_parameter_design.md

```diff
@@ -109,17 +109,28 @@ Each parameter consists of a key and a value.
 The key is a string value.
 The value can be one of the following datatypes:
 
-- `float64`
+- `bool`
 - `int64`
+- `float64`
 - `string`
-- `bool`
-- `bytes[]`
+- `byte[]`
```

Comment:
If there is a `byte[]` type should there be a `byte` type too?

The current version of the idl article mentions `byte` as a candidate for removal (see http://design.ros2.org/articles/interface_definition.html#primitive-field-types). I don't think that is reasonable but I just wanted mentioned it here.

Reply:
Interesting. I would argue that having `byte` arrays is a nice feature, which would necessitate that we keep `byte`. I guess we could use the `int64[]` parameter type to load arrays of bytes, just as we might use `int64[]` to load parameters that are meant to be stored internally as `int32[]`.

This raises another question: do we need to explicitly permit arrays of every numerical primitive, or should we limit it to the 64-bit versions, and force users to cast them after the parameters are loaded?

Reply:
So I agree completely. I'm happy to support the 64-bit types and let users worry about how to interpret the parameters.

What's the verdict on whether or not `bytes[]` lives?